src/widgets/table/table.rs

```diff
@@ -1484,7 +1364,264 @@ mod tests {
+                    "               ", // row 2
+                    "               ", // row 3
+                ]),
+            );
+            // highlight_symbol always rendered even no selection is made
+            test(
+                HighlightSpacing::Always,
+                None,
+                Buffer::with_lines(vec![
+                    "   hello  world", // row 1
+                    "               ", // row 2
+                    "               ", // row 3
+                ]),
+            );
+
+            // no highlight_symbol rendered ever
+            test(
+                HighlightSpacing::Never,
+                Some(0),
+                Buffer::with_lines(vec![
+                    "hello  world   ", // row 1
+                    "               ", // row 2
+                    "               ", // row 3
+                ]),
+            );
+            // highlight_symbol rendered because selection is made
+            test(
+                HighlightSpacing::WhenSelected,
+                Some(0),
+                Buffer::with_lines(vec![
+                    ">>>hello  world", // row 1
+                    "               ", // row 2
+                    "               ", // row 3
+                ]),
+            );
+            // no highlight_symbol rendered because no selection is made
+            test(
+                HighlightSpacing::Always,
+                Some(0),
+                Buffer::with_lines(vec![
+                    ">>>hello  world", // row 1
+                    "               ", // row 2
+                    "               ", // row 3
+                ]),
+            );
+        }
+
+        #[test]
+        fn spacers_priority_constrained_with_column_spacing() {
+            fn test(
+                highlight_spacing: HighlightSpacing,
+                selection: Option<usize>,
+                expected: Buffer,
+            ) {
+                let table = Table::default()
+                    .rows(vec![Row::new(vec!["hello", "12345"])])
+                    .highlight_spacing(highlight_spacing)
+                    .highlight_symbol(">>>")
+                    .column_spacing(1);
+                let area = Rect::new(0, 0, 10, 3);
+                let mut buf = Buffer::empty(area);
+                if let Some(s) = selection {
+                    let mut state = TableState::default().with_selected(Some(s));
+                    StatefulWidget::render(table, area, &mut buf, &mut state);
+                    assert_buffer_eq!(buf, expected);
+                } else {
+                    Widget::render(table, area, &mut buf);
+                    assert_buffer_eq!(buf, expected);
+                }
+            }
+
+            // column spacing is prioritized over everything constraint
+            test(
+                HighlightSpacing::Never,
+                None,
+                Buffer::with_lines(vec![
+                    "hello 1234", // row 1
```

Comment:
Is the use of truncation in these tests significant with respect to the tested properties, if so call it out, if not avoid it as it puts two behaviors on display and it's difficult to understand which is the behavior that is being tested and which is incidental.